src/app/contacts/add-contact.component.ts

```diff
@@ -0,0 +1,64 @@
+import { Component, Output } from '@angular/core';
+import { FormGroup, FormBuilder, Validators } from '@angular/forms';
+import { Gender } from './shared/contact.model';
+import { IContact } from './shared/contact.model';
+import { ContactService } from './shared/contact.service';
+import { EventEmitter } from '@angular/core';
+import { nameExistsAsyncValidator } from './shared/name-exists-async.validator';
+import { emailExistsAsyncValidator } from './shared/email-exists-async.validator';
+
+@Component({
+  selector: 'app-add-contact',
+  templateUrl: './add-contact.component.html',
+  styleUrls: ['./add-contact.component.css']
+})
+
+export class AddContactComponent {
+  contactForm: FormGroup;
+  gender = Gender;
+  nameExits: boolean;
+  submitted = false;
+
+  @Output() addContact: EventEmitter<IContact> = new EventEmitter<IContact>();
+
+  constructor(private fb: FormBuilder,
+              private contactService: ContactService) {
+    this.initContactForm();
+  }
+
+  onSave() {
+    this.submitted = true;
+    this.nameExits = false;
+
+    if (this.contactForm.invalid){
+      return;
+    }
+
+    const contact: IContact = {
+       name: this.contactForm.controls.name.value,
+       gender: this.contactForm.controls.gender.value as Gender,
+       email: this.contactForm.controls.email.value,
+       phone: this.contactForm.controls.phone.value,
+    };
+
+    this.addContact.emit(contact);
+
+    // reset form back to the original;
+    this.initContactForm();
```

Comment:
Tule ne veš a je bil kontakt res shranjen in ne smeš brez preverjanj resetirati forme.

Reply:
Se strinjam. Razmišljam v smeri: v main-contacts.component.ts bi le ob uspelem vnosu (znotraj addContact subscribe funkcije) poklical reset forme na add-contact.components.ts.

Reply:
A lahko prosim ob priliki implementiraš to spremembo? Drugih ni treba implementirati.

Reply:
Bom uredil.

Reply:
Težava je odpravljena, izdelal sem nov Pull request: 
#4 Rešeno: #3: Če pride do napake pri dodajanju kontakta, se vnosna polja ohranijo